src/btree/bt_stat.c

```diff
@@ -86,11 +86,21 @@ __stat_tree_walk(WT_SESSION_IMPL *session)
     WT_STAT_SET(session, stats, btree_row_leaf, 0);
 
     next_walk = NULL;
-    while (
-      (ret = __wt_tree_walk(session, &next_walk, WT_READ_VISIBLE_ALL)) == 0 && next_walk != NULL) {
+
+    /*
+     * Pages read for statistics aren't "useful"; don't update the read generation of pages already
+     * in memory, and if a page is read, set its generation to a low value so it is evicted quickly.
+     * Same as with compact.
+     */
+    while ((ret = __wt_tree_walk(session, &next_walk,
+              WT_READ_NO_GEN | WT_READ_WONT_NEED | WT_READ_VISIBLE_ALL)) == 0 &&
```

Comment:
Nit: Alphabetize the flags please. Also fix the alphabetizing of the same ones in bt_compact.c and txn_rollback_to_stable.c.

Reply:
Done.